fdbserver/workloads/CoordinatorsForce.actor.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * CoordinatorsForce.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbclient/ClusterInterface.h"
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbclient/ManagementAPI.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "fdbrpc/simulator.h"
+#include "flow/Arena.h"
+#include "flow/Platform.h"
+#include "flow/Trace.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+struct CoordinatorsForceWorkload : TestWorkload {
+
+	int numExtra;
+	double testDelay;
+	CoordinatorsForceWorkload(WorkloadContext const& wcx) : TestWorkload(wcx) {
+		numExtra = getOption(options, LiteralStringRef("numExtra"), 1);
+		testDelay = getOption(options, LiteralStringRef("testDelay"), 30.0);
+	}
+
+	std::string description() const override { return "CoordinatorsForce"; }
+
+	ACTOR static Future<Void> _start(Database cx, CoordinatorsForceWorkload* self) {
+		state std::vector<NetworkAddress> newCoordinators;
+		state int protectedAdded = 0;
+		state int normalAdded = 0;
+
+		// Wait and let simulation run a little bit first
+		wait(delay(self->testDelay));
+
+		// Similar to the check in ConsistencyCheck::checkCoordinators,
+		// ensure that two coordinators do not have the same zoneID
+		state vector<ProcessData> workers = wait(getWorkers(cx));
```

Comment:
I think this only fetches live workers?  Do you have a way to check that you're actually changing the coordinators to one that's unavailable here?

If there isn't a way to reach in and grab this information from g_simulator, then it might just work to move this above the `wait(delay(self->testDelay))`, because at the very start of a test, attrition probably wouldn't have killed anything yet?

Reply:
I tried before to fetch directly from `g_simulator` before, but it didn't seem to play nicely at all with `ConsistencyCheck::checkCoordinators`, so I used the same `getWorkers` and `zoneId` filtering processes that are used there. I will attempt the latter solution of getting the workers before the delay first, and if that doesn't seem to do what we want, I may revisit gathering the info from `g_simulator` again